src/App.tsx

```diff
@@ -15,36 +16,120 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  name = 'name',
+  length = 'length',
+  default = '',
+}
+
+const getVisibleGoods = (
+  goods: string[],
+  sortField: SortType,
+  isReversed: boolean,
+): string[] => {
+  if (!sortField) {
+    return goods;
+  }
+
+  const sortedGoods = [...goods].sort((good1, good2) => {
+    switch (sortField) {
+      case SortType.name:
+        return good1.localeCompare(good2);
+      case SortType.length:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.default);
+  const [isReversed, setIsReversed] = useState(false);
+  const [visibleGoods, setVisibleGoods] = useState<string[]>(goodsFromServer);
+
+  useEffect(() => {
+    const sortedVisibleGoods = getVisibleGoods(
+      visibleGoods,
+      sortField,
+      isReversed,
+    );
+
+    setVisibleGoods(sortedVisibleGoods);
+  }, [visibleGoods, sortField, isReversed]);
```

Comment:
you don't need separate state for visibleGoods

General rule: if you can "calculate" your value based on other state or props values you don't need a separate state for it. Because it will be recalculated every time when state/props change - during re-render

So:

```suggestion
    const visibleGoods = getVisibleGoods(
      goodsFromServer,
      sortField,
      isReversed,
    );
```